File: capacity-club-api/src/auth/model/entity/credential.entity.ts
import { BaseEntity } from '@common/model/base.entity';
import { UniqueId, uniqueIdTransformer } from '@common/model/unique-id';
import { ApiProperty } from '@nestjs/swagger';
import { Exclude, Transform } from 'class-transformer';
import { Member } from 'domain-modules/member/model';
import { Column, Entity, JoinColumn, OneToOne, PrimaryColumn } from 'typeorm';

@Entity()
export class Credential extends BaseEntity {
  /**
   * Unique identifier for the crendential.
   * Uses a custom transformer for serialization.
   */
  @Transform(uniqueIdTransformer.to, { toClassOnly: true })
  @Transform(uniqueIdTransformer.from, { toPlainOnly: true })
  @PrimaryColumn({
    type: 'varchar',
  })
  credential_id: UniqueId;

  // Username for the credential
  @Column({ nullable: false, unique: true })
  username: string;

  // Password for the credential, excluded from plain serialization
  @Exclude({ toPlainOnly: true })
  @Column({ nullable: true })
  password: string;

  // Facebook hash associated with the credential
  @Column({ nullable: true, unique: false })
  facebookHash: string;

  // Google hash associated with the credential
  @Column({ nullable: true, unique: false })
  googleHash: string;

  // Flag indicating if the user is an admin
  @Column({ default: false })
  is_admin: boolean;

  // Flag indicating if the user is an admin
  @Column({ default: true })
  is_active: boolean;

  // Member associated with the credential
  @OneToOne(() => Member, { nullable: true, lazy: true })
  @JoinColumn({
    name: 'member_id_fk',
    referencedColumnName: 'member_id',
  })
  @ApiProperty({ type: () => Member })
  member: Promise<Member>;
}

File: capacity_club_mobile_app/lib/auth/domain/entity/credential_entity.dart
import 'package:capacity_club_mobile_app/core/model/entities/unique_id.dart';
import 'package:capacity_club_mobile_app/domain-features/domain/member/entity/member_entity.dart';

class CredentialEntity {
  final UniqueId credential_id;
  final String username;
  final String googleHash;
  final String facebookHash;
  final bool is_admin;
  final bool is_active;
  final MemberEntity member;
  final bool is_empty;

  const CredentialEntity(
      {required this.credential_id,
      required this.username,
      required this.googleHash,
      required this.facebookHash,
      required this.is_admin,
      required this.is_active,
      required this.member,
      required this.is_empty});

  factory CredentialEntity.empty() {
    return CredentialEntity(
        credential_id: UniqueId(''),
        username: '',
        googleHash: '',
        facebookHash: '',
        is_admin: false,
        is_active: false,
        member: MemberEntity.empty(),
        is_empty: true);
  }
}
